### PR TITLE
fix golint warnings/errors on pkg/system and pkg/stdcopy

### PR DESCRIPTION
**Pull request by WeiZhang555**
Part of #14756

Signed-off-by: Zhang Wei zhangwei555@huawei.com

### CODE REVIEW

File: pkg/system/stat.go [hunk 1]
```diff
@@ -17,30 +17,37 @@ type Stat_t struct {
...
 }
 
-func (s Stat_t) Mode() uint32 {
+// Mode returns file's permission mode
+func (s StatT) Mode() uint32 {
 	return s.mode
 }
 
-func (s Stat_t) Uid() uint32 {
+// UID return file's user id of owner
```
Comment (vdemeester) [inline pkg/system/stat.go]: s/return/returns/

File: pkg/system/stat.go [hunk 2]
```diff
@@ -17,30 +17,37 @@ type Stat_t struct {
...
+func (s StatT) UID() uint32 {
 	return s.uid
 }
 
-func (s Stat_t) Gid() uint32 {
+// Gid return file's group id of owner
```
Comment (vdemeester) [inline pkg/system/stat.go]: s/return/returns/

File: pkg/system/stat.go [hunk 3]
```diff
@@ -17,30 +17,37 @@ type Stat_t struct {
...
+func (s StatT) Gid() uint32 {
 	return s.gid
 }
 
-func (s Stat_t) Rdev() uint64 {
+// Rdev return file's device ID (if it's special file)
```
Comment (vdemeester) [inline pkg/system/stat.go]: s/return/returns/

File: pkg/system/stat.go [hunk 4]
```diff
@@ -17,30 +17,37 @@ type Stat_t struct {
...
+func (s StatT) Rdev() uint64 {
 	return s.rdev
 }
 
-func (s Stat_t) Size() int64 {
+// Size return file's size
```
Comment (vdemeester) [inline pkg/system/stat.go]: s/return/returns/

File: pkg/system/stat.go [hunk 5]
```diff
@@ -17,30 +17,37 @@ type Stat_t struct {
...
+func (s StatT) Size() int64 {
 	return s.size
 }
 
-func (s Stat_t) Mtim() syscall.Timespec {
+// Mtim return file's last modification time
```
Comment (vdemeester) [inline pkg/system/stat.go]: s/return/returns/

File: pkg/system/stat.go [hunk 6]
```diff
@@ -17,30 +17,37 @@ type Stat_t struct {
...
+func (s StatT) Mtim() syscall.Timespec {
 	return s.mtim
 }
 
-func (s Stat_t) GetLastModification() syscall.Timespec {
+// GetLastModification return file's last modification time
```
Comment (vdemeester) [inline pkg/system/stat.go]: s/return/returns/

**Comment by gdevillele** [comment]
You need a period at the end of the sentences for the func comments.

**Comment by tiborvass** [comment]
@WeiZhang555 sorry needs another rebase :(

**Comment by WeiZhang555** [comment]
Refreshed, thank you for help reviewing!
ping @tiborvass @gdevillele @vdemeester

**Comment by tiborvass** [comment]
LGTM

**Comment by WeiZhang555** [comment]
It's strange that it looks like I missed some file, golint `pkg/system` report nothing but golint `pkg/system/events_windows.go` report warnings, maybe this is a `golint` bug?

**Comment by WeiZhang555** [comment]
Updated, please help review @all-maintainers  :)

**Comment by vdemeester** [comment]
> It's strange that it looks like I missed some file, golint pkg/system report nothing but golint pkg/system/events_windows.go report warnings, maybe this is a golint bug?

@WeiZhang555 ``golint`` do not look for other platform go files (or at least the ``sthg_else.go``). That's why we do ``golint $package/*.go`` in ``validate-lint``, to make sure we don't miss them :stuck_out_tongue_closed_eyes: .

File: pkg/system/meminfo_unsupported.go [hunk 7]
```diff
@@ -2,6 +2,7 @@
 
 package system
 
+// ReadMemInfo doesn't support platforms other than linux and windows
```
Comment (vdemeester) [inline pkg/system/meminfo_unsupported.go]: s/doesn't support/is not supporty by/
but it's a nit :blush:

File: pkg/system/mknod.go [hunk 8]
```diff
@@ -12,6 +12,8 @@ func Mknod(path string, mode uint32, dev int) error {
 	return syscall.Mknod(path, mode, dev)
 }
 
+// Mkdev is used to build the value of dev which specifies major and minor number of the
```
Comment (vdemeester) [inline pkg/system/mknod.go]: s/of dev/of linux devices (in /dev/)/
or something like that maybe :blush:

File: pkg/system/mknod_windows.go [hunk 9]
```diff
@@ -2,10 +2,12 @@
 
 package system
 
+// Mknod is not implemented on Windows
```
Comment (vdemeester) [inline pkg/system/mknod_windows.go]: Add a period at the end :sweat_smile:

File: pkg/system/mknod_windows.go [hunk 10]
```diff
@@ -2,10 +2,12 @@
...
 func Mknod(path string, mode uint32, dev int) error {
 	return ErrNotSupportedPlatform
 }
 
+// Mkdev is not implemented on Windows
```
Comment (vdemeester) [inline pkg/system/mknod_windows.go]: Add a period at the end :sweat_smile:

File: pkg/system/stat_windows.go [hunk 11]
```diff
@@ -15,22 +17,27 @@ type Stat_t struct {
 	isDir   bool
 }
 
-func (s Stat_t) Name() string {
+// Name return file's name.
```
Comment (vdemeester) [inline pkg/system/stat_windows.go]: s/return/returns/

File: pkg/system/stat_windows.go [hunk 12]
```diff
@@ -15,22 +17,27 @@ type Stat_t struct {
...
+func (s StatT) Name() string {
 	return s.name
 }
 
-func (s Stat_t) Size() int64 {
+// Size return file's size.
```
Comment (vdemeester) [inline pkg/system/stat_windows.go]: s/return/returns/

File: pkg/system/stat_windows.go [hunk 13]
```diff
@@ -15,22 +17,27 @@ type Stat_t struct {
...
+func (s StatT) Size() int64 {
 	return s.size
 }
 
-func (s Stat_t) Mode() os.FileMode {
+// Mode return file's permission mode.
```
Comment (vdemeester) [inline pkg/system/stat_windows.go]: s/return/returns/

File: pkg/system/stat_windows.go [hunk 14]
```diff
@@ -15,22 +17,27 @@ type Stat_t struct {
...
+func (s StatT) Mode() os.FileMode {
 	return s.mode
 }
 
-func (s Stat_t) ModTime() time.Time {
+// ModTime return file's last modification time.
```
Comment (vdemeester) [inline pkg/system/stat_windows.go]: s/return/returns/

File: pkg/system/umask_windows.go [hunk 15]
```diff
@@ -2,6 +2,7 @@
 
 package system
 
+// Umask doesn't support windows platform now.
```
Comment (vdemeester) [inline pkg/system/umask_windows.go]: "Umask is not supported on the windows platform"

File: pkg/system/utimes_freebsd.go [hunk 16]
```diff
@@ -5,6 +5,9 @@ import (
 	"unsafe"
 )
 
+// LUtimesNano is used to change access and modification time of path
```
Comment (vdemeester) [inline pkg/system/utimes_freebsd.go]: s/path/specified path/ maybe :blush:

File: pkg/system/utimes_freebsd.go [hunk 17]
```diff
@@ -19,6 +22,8 @@ func LUtimesNano(path string, ts []syscall.Timespec) error {
 	return nil
 }
 
+// UtimesNano is used to change access and modification time of path.
+// it can't be used for symbol link file.
```
Comment (vdemeester) [inline pkg/system/utimes_freebsd.go]: s/it/It/

File: pkg/system/utimes_linux.go [hunk 18]
```diff
@@ -5,10 +5,13 @@ import (
 	"unsafe"
 )
 
+// LUtimesNano is used to change access and modification time of path
```
Comment (vdemeester) [inline pkg/system/utimes_linux.go]: Same as above in ``utimes_freebsd.go``

File: pkg/system/utimes_linux.go [hunk 19]
```diff
@@ -16,13 +19,15 @@ func LUtimesNano(path string, ts []syscall.Timespec) error {
...
-	if _, _, err := syscall.Syscall6(syscall.SYS_UTIMENSAT, uintptr(AT_FDCWD), uintptr(unsafe.Pointer(_path)), uintptr(unsafe.Pointer(&ts[0])), uintptr(AT_SYMLINK_NOFOLLOW), 0, 0); err != 0 && err != syscall.ENOSYS {
+	if _, _, err := syscall.Syscall6(syscall.SYS_UTIMENSAT, uintptr(atFdCwd), uintptr(unsafe.Pointer(_path)), uintptr(unsafe.Pointer(&ts[0])), uintptr(atSymLinkNoFollow), 0, 0); err != 0 && err != syscall.ENOSYS {
 		return err
 	}
 
 	return nil
 }
 
+// UtimesNano is used to change access and modification time of path.
+// it can't be used for symbol link file.
```
Comment (vdemeester) [inline pkg/system/utimes_linux.go]: s/it/It/

File: pkg/system/utimes_unsupported.go [hunk 20]
```diff
@@ -4,10 +4,12 @@ package system
 
 import "syscall"
 
+// LUtimesNano doesn't support platforms other than linux, freebsd and darwin.
```
Comment (vdemeester) [inline pkg/system/utimes_unsupported.go]: s/doesn't support/is not supported on/

File: pkg/system/utimes_unsupported.go [hunk 21]
```diff
@@ -4,10 +4,12 @@ package system
...
 func LUtimesNano(path string, ts []syscall.Timespec) error {
 	return ErrNotSupportedPlatform
 }
 
+// UtimesNano doesn't support platforms other than linux, freebsd and darwin.
```
Comment (vdemeester) [inline pkg/system/utimes_unsupported.go]: same as above

File: pkg/system/xattrs_linux.go [hunk 22]
```diff
@@ -5,7 +5,9 @@ import (
 	"unsafe"
 )
 
-// Returns a nil slice and nil error if the xattr is not set
+// Lgetxattr retrieves the value of the extended attribute identified by attr
+// and associated with the given path in the file system.
+// it will returns a nil slice and nil error if the xattr is not set.
```
Comment (vdemeester) [inline pkg/system/xattrs_linux.go]: s/it/It/

File: pkg/system/xattrs_unsupported.go [hunk 23]
```diff
@@ -2,10 +2,12 @@
 
 package system
 
+// Lgetxattr doesn't support platforms other than linux.
```
Comment (vdemeester) [inline pkg/system/xattrs_unsupported.go]: Same as above :blush:.

**Comment by vdemeester** [comment]
@WeiZhang555 There is a few compilation error still :sweat_smile:.
```
pkg/system/lstat.go:18: cannot use <T> as type *StatT in return argument
pkg/system/stat_freebsd.go:8: undefined: Stat_t
pkg/system/stat_freebsd.go:9: undefined: Stat_t
pkg/system/stat_freebsd.go:21: undefined: Stat_t
```

**Comment by WeiZhang555** [comment]
@vdemeester Updated again. Thank you for your kind and patient advice! :smile: 
Let's see how it works this time.

**Comment by jeffdm** [comment]
Should `pkg/archive` be added to `hack/make/validate-lint`?

**Comment by tiborvass** [comment]
@jeffdm No because this PR doesn't address `pkg/archive`. The only changes in `pkg/archive` are to rename a variable from `pkg/system`.

**Comment by icecrime** [comment]
@vdemeester @gdevillele PTAL!

File: pkg/system/meminfo_unsupported.go [hunk 24]
```diff
@@ -2,6 +2,7 @@
 
 package system
 
+// ReadMemInfo is not supported platforms other than linux and windows
```
Comment (vdemeester) [inline pkg/system/meminfo_unsupported.go]: small nit, a period here :blush:.

**Comment by vdemeester** [comment]
LGTM :wink:

**Comment by WeiZhang555** [comment]
@vdemeester period added, thank you! :laughing:

**Comment by WeiZhang555** [comment]
Rebased

**Comment by gdevillele** [comment]
`$ make test` failed here.

**Comment by WeiZhang555** [comment]
All green now.
Ping @gdevillele @vdemeester @tiborvass @icecrime

**Comment by gdevillele** [comment]
LGTM

**Comment by WeiZhang555** [comment]
Ping @gdevillele @vdemeester @tiborvass @icecrime
Any progress about this? :smile:

File: hack/make/validate-lint [hunk 25]
```diff
@@ -73,11 +73,14 @@ packages=(
...
 	pkg/sockets
+	pkg/stdcopy
 	pkg/streamformatter
 	pkg/stringid
 	pkg/stringutils
-	pkg/sysinfo
 	pkg/symlink
+	pkg/sysinfo
+	pkg/system
+	pkg/systemd
```
Comment (vdemeester) [inline hack/make/validate-lint]: Could you remove this one ? ``pkg/systemd`` is not here anymore (and thus is not covered by the PR :sweat_smile:)

Comment (WeiZhang555) [inline hack/make/validate-lint]: Oops, some rebase mistake, will remove it.

**Comment by WeiZhang555** [comment]
updated :)

**Comment by vdemeester** [comment]
@WeiZhang555 you're gonna need yet another rebase (to fix the experimental build). sorry :sweat_smile:  (hopefully this is the last one :wink: )

**Comment by WeiZhang555** [comment]
OK, updated. :smile:

**Comment by WeiZhang555** [comment]
Ping @gdevillele @vdemeester @tiborvass @icecrime

**Comment by vdemeester** [comment]
still LGTM for me :wink:

**Comment by gdevillele** [comment]
LGTM

**Comment by jfrazelle** [comment]
LGTM

**Pull request merged**